File: src/main/java/me/tomasan7/jecnadesktop/ui/component/SidebarLinkElement.java
package me.tomasan7.jecnadesktop.ui.component;

import javafx.fxml.FXML;
import javafx.fxml.FXMLLoader;
import javafx.scene.control.Label;
import javafx.scene.image.Image;
import javafx.scene.image.ImageView;

import java.awt.*;
import java.io.IOException;
import java.net.URI;

public class SidebarLinkElement extends SidebarElement
{
	@FXML
	private ImageView linkIcon;
	@FXML
	private String link;

	public SidebarLinkElement ()
	{
		FXMLLoader fxmlLoader = new FXMLLoader(getClass().getResource("/ui/component/SideBarLinkElement.fxml"));

		try
		{
			fxmlLoader.setRoot(this);
			fxmlLoader.setController(this);
			fxmlLoader.load();
		}
		catch (IOException exception)
		{
			throw new RuntimeException(exception);
		}

		setOnMouseClicked(event ->
						  {
							  if (!Desktop.isDesktopSupported())
								  return;

							  Desktop desktop = Desktop.getDesktop();

							  try
							  {
								  desktop.browse(URI.create(link));
							  }
							  catch (IOException e)
							  {
								  e.printStackTrace();
							  }
						  });
	}

	public ImageView getLinkIcon ()
	{
		return linkIcon;
	}

	public void setLinkIcon (ImageView linkIcon)
	{
		this.linkIcon = linkIcon;
	}

	public String getLink ()
	{
		return link;
	}

	public void setLink (String link)
	{
		this.link = link;
	}
}

File: src/main/kotlin/me/tomasan7/jecnadesktop/ui/component/SidebarLinkElement.kt
package me.tomasan7.jecnadesktop.ui.component

import javafx.event.EventHandler
import javafx.fxml.FXML
import javafx.fxml.FXMLLoader
import java.awt.Desktop
import java.net.URI

class SidebarLinkElement : SidebarElement()
{
    @FXML
    lateinit var linkIcon: IconView

    @FXML
    lateinit var link: String

    init
    {
        val fxmlLoader = FXMLLoader(javaClass.getResource("/ui/component/SidebarLinkElement.fxml"))

        with(fxmlLoader)
        {
            setRoot(this@SidebarLinkElement)
            setController(this@SidebarLinkElement)
            load<Any>()
        }

        onMouseClicked = EventHandler {
            if (!Desktop.isDesktopSupported())
                return@EventHandler

            val desktop = Desktop.getDesktop()

            desktop.browse(URI.create(link))
        }
    }
}